src/pip/_internal/commands/cache.py

```diff
@@ -183,7 +183,28 @@ def remove_cache_items(self, options: Values, args: List[Any]) -> None:
         for filename in files:
             os.unlink(filename)
             logger.verbose("Removed %s", filename)
+
+        http_dirs = filesystem.subdirs_with_no_files(self._cache_dir(options, "http"))
+        wheel_dirs = filesystem.subdirs_with_no_files(
+            self._cache_dir(options, "wheels")
+        )
+        dirs = list(http_dirs) + list(wheel_dirs)
```

Comment:
```suggestion
        dirs = [*http_dirs, *wheel_dirs]
```

This should reduce memory usage a bit it avoids building the intermediate lists.